/spec/staking/ add copy on Slashing and Delegator Shares (#5020)
cosmos/cosmos-sdk: docs/spec/staking/01_state.md

## Validator

Validators can have one of three statuses

- `unbonded`: The validator is not in the active set. They cannot sign blocks and do not earn
rewards. They can receive delegations. Unbonding from an `unbonded` validator is immediate.

> - `bonded`: The validator has sufficient bonded tokens to join the active set. They are signing

Reviewer: Wording is a bit weird? The validator is already part of the active set.

blocks and receiving rewards. They can receive further delegations. They can be slashed for
misbehavior. Delegators to this validator who unbond their delegation must wait the duration of the
UnbondingTime, a chain-specific param.

> - `unbonding`: The validator has initiated an unbonding of their self delegation. They must wait the

Reviewer: There are other means in which a validator can start to unbond (e.g. being kicked out of the active set).

## Validator

Validators can have one of three statuses

- `Unbonded`: The validator is not in the active set. They cannot sign blocks and do not earn 

>   rewards. They can receive delegations. Unbonding from an `Unbonded` validator is immediate.

Reviewer 2: I'd to confirm that 
> Unbonding from an `Unbonded` validator is immediate.

is true within the code. This thinking has changed a couple of times

>   rewards. They can receive delegations. Unbonding from an `Unbonded` validator is immediate.

Reviewer 2: AFAICT this is _not_ true within the code, the delegator always must wait the unbonding period independent of whether or not the validator is not-bonded. See https://github.com/cosmos/cosmos-sdk/blob/master/x/staking/keeper/delegation.go#L653

- `Bonded`": Once the validator receives sufficient bonded tokens they join the active set and 

>   their status is updated to `Bonded`. They are signing blocks and receiving rewards. They can 

Author: oh wow sorry dont know why I didnt see that

  receive further delegations. They can be slashed for misbehavior. Delegators to this validator 

>   who unbond their delegation must wait the duration of the UnbondingTime, a chain-specific param. 

Reviewer 2: ```suggestion
  who unbond their delegation must wait the duration of the UnbondingTime, a chain-specific param during which point they are still slashable for offences of the source validator if those offences were committed during a period of time where there tokens were bonded. 
```

- `Unbonding`: When a validator leaves the active set, either by choice or due to slashing or 

>   tombstoning, an unbonding of their delegations begins. They and their must wait the UnbondingTime 

Reviewer 2: ```suggestion
  tombstoning, an unbonding of all their delegations begin. And all delegations must wait the UnbondingTime 
```

}
```

### Delegator Shares

When one Delegates tokens to a Validator they are issued a number of delegator shares based on a
dynamic exchange rate, calculated as follows from the total number of tokens delegated to the
validator and the number of shares issued so far:

`Shares per Token = validator.TotalShares() / validator.Tokens()`

> Only the number of shares is received is stored on the DelegationEntry. When a delegator then

Reviewer 2: ```suggestion
Only the number of shares received is stored on the DelegationEntry. When a delegator then
```
